src/main/java/repo/impl/PostgresSessionRepo.java

```diff
@@ -0,0 +1,129 @@
+                    "restoreXml, username, initLang, sequenceId, " +
+                    "domain, postUrl, sessionData) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)");
+            this.jdbcTemplate.update(query,  new Object[] {session.getId(), session.getInstanceXml(), session.getFormXml(),
+                    session.getRestoreXml(), session.getUsername(), session.getInitLang(), session.getSequenceId(),
+                    session.getDomain(), session.getPostUrl(), sessionDataBytes}, new int[] {
+                    Types.VARCHAR, Types.VARCHAR, Types.VARCHAR, Types.VARCHAR, Types.VARCHAR,
+                    Types.VARCHAR, Types.VARCHAR, Types.VARCHAR, Types.VARCHAR, Types.BINARY});
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public SerializableFormSession find(String id) {
+        String sql = replaceTableName("SELECT * FROM %s WHERE id = ?");
+        return jdbcTemplate.queryForObject(sql, new Object[] {id}, new SessionMapper());
+    }
+
+    @Override
+    public List<SerializableFormSession> findUserSessions(String username) {
+        List<SerializableFormSession> sessions = this.jdbcTemplate.query(
+                replaceTableName("SELECT * FROM %s WHERE username = ?"),
+                new Object[] {username},
+                new SessionMapper());
+        return sessions;
+    }
+
+    @Override
+    public void delete(String id) {
+        this.jdbcTemplate.update(replaceTableName("DELETE FROM %s WHERE id = ?"), id);
+    }
+
+    // helper class for mapping a db row to a serialized session
+    private static final class SessionMapper implements RowMapper<SerializableFormSession> {
+
+        public SerializableFormSession mapRow(ResultSet rs, int rowNum) throws SQLException {
+
+            SerializableFormSession session = new SerializableFormSession();
+            session.setId(rs.getString("id"));
+            session.setInstanceXml(rs.getString("instanceXml"));
+            session.setFormXml(rs.getString("formXml"));
+            session.setRestoreXml(rs.getString("restoreXml"));
+            session.setUsername(rs.getString("username"));
+            session.setInitLang(rs.getString("initLang"));
+            session.setSequenceId(Integer.parseInt(rs.getString("sequenceId")));
+            session.setDomain(rs.getString("domain"));
+            session.setPostUrl(rs.getString("postUrl"));
+
+            byte[] st = (byte[]) rs.getObject("sessionData");
+
+            if(st != null) {
+                ByteArrayInputStream byteInputStream = new ByteArrayInputStream(st);
+                ObjectInputStream objectInputStream;
+                try {
+                    objectInputStream = new ObjectInputStream(byteInputStream);
+                    Map<String, String> sessionData = (HashMap) objectInputStream.readObject();
+                    session.setSessionData(sessionData);
+                } catch (IOException e) {
```

Comment:
can you leave a comment as to when we'd see these and why we don't care about them?